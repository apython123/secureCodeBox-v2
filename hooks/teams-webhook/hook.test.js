const { handle, axios } = require("./hook");

beforeEach(() => {
  axios.post.mockClear();
});

test("should send a post request to the url when fired", async () => {
  const findings = [];

  const getFindings = async () => findings;

  const scan = {
    metadata: {
      uid: "09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
      name: "demo-scan-1601086432",
      namespace: "my-scans",
      labels: {
        company: "iteratec",
        "attack-surface": "external",
      },
    },
    spec: {
      scanType: "Nmap",
      parameters: ["-Pn", "localhost"],
    },
    status: {
      findingDownloadLink: "https://my-secureCodeBox-instance.com/scan-b9as-sdweref--sadf-asdfsdf-dasdgf-asdffdsfa7/findings.json",
      findings: {
          "categories": {
              "A Client Error response code was returned by the server": 1,
              "Information Disclosure - Sensitive Information in URL": 1,
              "Strict-Transport-Security Header Not Set": 1
          },
          count: 3,
          severities: {
            "high": 10,
            "medium": 5,
            "low": 2,
            "informational": 1,
          }
      },
      finishedAt: "2020-05-25T02:38:13Z",
      rawResultDownloadLink: "https://my-secureCodeBox-instance.com/scan-blkfsdg-sdgfsfgd-sfg-sdfg-dfsg-gfs98-e8af2172caa7/zap-results.json?Expires=1601691232",
      rawResultFile: "zap-results.json",
      rawResultType: "zap-json",
      state: "Done"
    }
  };

  const webhookUrl = "http://example.com/foo/bar";

  await handle({ getFindings, scan, webhookUrl });

  expect(axios.post).toMatchSnapshot();
});
