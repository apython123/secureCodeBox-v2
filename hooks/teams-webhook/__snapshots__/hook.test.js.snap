// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should send a post request to the url when fired 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "http://example.com/foo/bar",
      Object {
        "findings": Array [],
        "paylod": Object {
          "@context": "https://schema.org/extensions",
          "@type": "MessageCard",
          "potentialAction": Array [
            Object {
              "@type": "OpenUri",
              "name": "Open Dashboard",
              "targets": Array [
                Object {
                  "os": "default",
                  "uri": "https://your-dashboard.url/",
                },
              ],
            },
            Object {
              "@type": "OpenUri",
              "name": "Show Results in Dashboard",
              "targets": Array [
                Object {
                  "os": "default",
                  "uri": "https://your-dashboard.url/",
                },
              ],
            },
          ],
          "sections": Array [
            Object {
              "activityImage": "https://raw.githubusercontent.com/secureCodeBox/securecodebox.github.io/gh-source/static/Favicon.png",
              "activitySubtitle": "Finished at 2020-05-25T02:38:13Z",
              "activityTitle": "Scheduled scan: **'demo-scan-1601086432'**",
              "facts": Array [
                Object {
                  "name": "High:",
                  "value": "10",
                },
                Object {
                  "name": "Medium:",
                  "value": "5",
                },
                Object {
                  "name": "Low:",
                  "value": "2",
                },
                Object {
                  "name": "Informational",
                  "value": "1",
                },
              ],
              "startGroup": true,
              "text": "__Findings Severity Overview:__",
            },
            Object {
              "facts": Array [
                Object {
                  "name": "Open Ports:",
                  "value": "3",
                },
                Object {
                  "name": "Hosts:",
                  "value": "8",
                },
              ],
              "text": "__Findings Category Overview:__",
            },
          ],
          "summary": "Scan 09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
          "themeColor": "0078D7",
          "title": "New security scan Nmap results are available!",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
